### PR TITLE
drafted user controller, edited routes to accept module exports

### PR DESCRIPTION
**Pull request by mcp001**
1st draft of User controller.  Was unsure about formatting bcrypt/cookie with async/await and TS, so went with more traditional JS formatting for now to allow functionality.  Some typing was added.  User routes edited to accept module exports.  Feel free to edit.

### CODE REVIEW

File: server/controllers/user.controller.ts [hunk 1]
```diff
@@ -1,8 +1,109 @@
...
+// const register = async (req: Request, res: Response) => {}
+// const login = async (req: Request, res: Response) => {}
+// const logout = async (req: Request, res: Response) => {}
+// const getOneUser = async (req: Request, res: Response) => {}
+
+// export default { register, login, logout, getOneUser }
+
 import { Request, Response } from "express";
+import User from "../models/user.model"
+const bcrypt = require('bcrypt');
```
Comment (b-chang) [inline server/controllers/user.controller.ts]: if you want to follow ES6 syntax we can do the following

`import bcrypt from "bcrypt";`

File: server/controllers/user.controller.ts [hunk 2]
```diff
@@ -1,8 +1,109 @@
+// import { IUser } from "../interfaces/user.interface";
```
Comment (b-chang) [inline server/controllers/user.controller.ts]: can you remove these commented out code

File: server/controllers/user.controller.ts [hunk 3]
```diff
@@ -1,8 +1,109 @@
...
+                console.log("password is valid");
+
+                res.cookie(
+                  "usertoken",
+                  jwt.sign(
+                    {
+                      id: userRecord._id,
+                      email: userRecord.email,
+                    },
+                    process.env.JWT_SECRET,
```
Comment (b-chang) [inline server/controllers/user.controller.ts:40]: does the `process.env.JWT_SECRET` work for you? When i tried last night I had to bring in the dotenv module and use `dotenv.config()` to get it to work so im curious if i was doing something wrong

**Review by b-chang — CHANGES_REQUESTED** [review]

File: server/controllers/user.controller.ts [hunk 4]
```diff
@@ -1,8 +1,109 @@
...
+      .then((newUser) => {
+        console.log(newUser);
+        console.log("registration successful");
+        res.json({
+          successMessage: "Thank you for registering!",
+          user: newUser,
+        });
+      })
+      .catch((err: unknown) => {
+        console.log("registration unsuccessful");
```
Comment (b-chang) [inline server/controllers/user.controller.ts]: Don't forget to remove the console.logs as per kishan mentioned

**Review by b-chang — CHANGES_REQUESTED** [review]

File: server/controllers/user.controller.ts (hunk 1 above)
Comment (mcp001) [reply server/controllers/user.controller.ts]: done

File: server/controllers/user.controller.ts (hunk 2 above)
Comment (mcp001) [reply server/controllers/user.controller.ts]: done

File: server/controllers/user.controller.ts (hunk 3 above)
Comment (mcp001) [reply server/controllers/user.controller.ts:40]: we'll check this later on

File: server/controllers/user.controller.ts (hunk 4 above)
Comment (mcp001) [reply server/controllers/user.controller.ts]: done, thanks

File: server/controllers/user.controller.ts (hunk 3 above)
Comment (pmar0) [reply server/controllers/user.controller.ts:40]: checked and it works fine if dotenv import and `dotenv.config()` is in the `server.ts`

**Review by pmar0 — APPROVED** [review]
pushing this one through to move forward. Can tweak imports and such later, if needed.

**Pull request merged**